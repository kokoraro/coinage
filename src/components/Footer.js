import React from 'react';

const Footer = () => (
    <footer>
        <p>
            coded by
            <a href="https://www.charlietaylorcoder.com">charlie taylor</a>
        </p>
        <div className="footer-icons">
            <a href="http://twitter.com/charlietcoder"><i className="fa fa-twitter" /></a>
            <a href="http://www.instagram.com/charlietcoder"><i className="fa fa-instagram" /></a>
            <a href="https://codepen.io/chazmcgrill"><i className="fa fa-codepen" /></a>
            <a href="https://github.com/chazmcgrill"><i className="fa fa-github" /></a>
        </div>
        <p>
            uses
            <a href="https://www.cryptocompare.com/api/">cryptocompare api</a>
        </p>
    </footer>
);

export default Footer;
