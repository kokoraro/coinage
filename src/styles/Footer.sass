footer
    display: flex
    justify-content: space-between
    padding: 20px 0
    background-color: pink
    align-items: center
    flex-direction: column
      
    p
        margin: 10px 0
        font-size: 0.6rem

.footer-icons
    width: 180px
    display: grid
    grid-template-columns: repeat(4, 1fr)
    text-align: center
    font-size: 1.3rem

@media screen and (min-width: 700px)
    footer
        flex-direction: row
        padding: 40px 20px

        p
            margin: 0
            font-size: 0.6rem