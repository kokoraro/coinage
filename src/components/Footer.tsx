import React from 'react';
import '../styles/Footer.sass';

interface IconProps {
    url: string;
    icon: string;
}

const Icon = ({ url, icon }: IconProps) => {
    const iconClass = `fab fa-${icon}`;
    return (
        <a href={url}>
            <i className={iconClass} />
        </a>
    );
};

const Footer = () => (
    <footer>
        <p>coded by<a href="https://www.charlietaylorcoder.com"> charlie taylor</a></p>
        <div className="footer-icons">
            <Icon url="http://twitter.com/charlietcoder" icon="twitter" />
            <Icon url="http://www.instagram.com/charlietcoder" icon="instagram" />
            <Icon url="https://codepen.io/chazmcgrill" icon="codepen" />
            <Icon url="https://github.com/chazmcgrill" icon="github" />
        </div>
        <p>uses<a href="https://www.cryptocompare.com/api/"> cryptocompare api</a></p>
    </footer>
);

export default Footer;
