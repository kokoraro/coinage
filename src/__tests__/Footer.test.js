import React from "react";
import { shallow } from "enzyme";
import Footer from "../components/Footer";

const app = shallow(<Footer />);

describe("Footer", () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});
