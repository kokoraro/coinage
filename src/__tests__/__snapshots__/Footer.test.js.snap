// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          designed and coded by 
          <a
            href="https://www.charlietaylorcoder.com"
          >
            charlie taylor
          </a>
        </p>,
        <div
          className="footer-icons"
        >
          <a
            href="http://twitter.com/charlietcoder"
          >
            <i
              className="fa fa-twitter"
            />
          </a>
          <a
            href="http://www.instagram.com/charlietcoder"
          >
            <i
              className="fa fa-instagram"
            />
          </a>
          <a
            href="http://www.behance.net/chazhurricane"
          >
            <i
              className="fa fa-behance"
            />
          </a>
        </div>,
        <p>
          data from 
          <a
            href="https://www.cryptocompare.com/api/"
          >
            cryptocompare api
          </a>
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "designed and coded by ",
            <a
              href="https://www.charlietaylorcoder.com"
            >
              charlie taylor
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "designed and coded by ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "charlie taylor",
              "href": "https://www.charlietaylorcoder.com",
            },
            "ref": null,
            "rendered": "charlie taylor",
            "type": "a",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="http://twitter.com/charlietcoder"
            >
              <i
                className="fa fa-twitter"
              />
            </a>,
            <a
              href="http://www.instagram.com/charlietcoder"
            >
              <i
                className="fa fa-instagram"
              />
            </a>,
            <a
              href="http://www.behance.net/chazhurricane"
            >
              <i
                className="fa fa-behance"
              />
            </a>,
          ],
          "className": "footer-icons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-twitter"
              />,
              "href": "http://twitter.com/charlietcoder",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-twitter",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-instagram"
              />,
              "href": "http://www.instagram.com/charlietcoder",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-instagram",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-behance"
              />,
              "href": "http://www.behance.net/chazhurricane",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-behance",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "data from ",
            <a
              href="https://www.cryptocompare.com/api/"
            >
              cryptocompare api
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "data from ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "cryptocompare api",
              "href": "https://www.cryptocompare.com/api/",
            },
            "ref": null,
            "rendered": "cryptocompare api",
            "type": "a",
          },
        ],
        "type": "p",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            designed and coded by 
            <a
              href="https://www.charlietaylorcoder.com"
            >
              charlie taylor
            </a>
          </p>,
          <div
            className="footer-icons"
          >
            <a
              href="http://twitter.com/charlietcoder"
            >
              <i
                className="fa fa-twitter"
              />
            </a>
            <a
              href="http://www.instagram.com/charlietcoder"
            >
              <i
                className="fa fa-instagram"
              />
            </a>
            <a
              href="http://www.behance.net/chazhurricane"
            >
              <i
                className="fa fa-behance"
              />
            </a>
          </div>,
          <p>
            data from 
            <a
              href="https://www.cryptocompare.com/api/"
            >
              cryptocompare api
            </a>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "designed and coded by ",
              <a
                href="https://www.charlietaylorcoder.com"
              >
                charlie taylor
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "designed and coded by ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "charlie taylor",
                "href": "https://www.charlietaylorcoder.com",
              },
              "ref": null,
              "rendered": "charlie taylor",
              "type": "a",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="http://twitter.com/charlietcoder"
              >
                <i
                  className="fa fa-twitter"
                />
              </a>,
              <a
                href="http://www.instagram.com/charlietcoder"
              >
                <i
                  className="fa fa-instagram"
                />
              </a>,
              <a
                href="http://www.behance.net/chazhurricane"
              >
                <i
                  className="fa fa-behance"
                />
              </a>,
            ],
            "className": "footer-icons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-twitter"
                />,
                "href": "http://twitter.com/charlietcoder",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-twitter",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-instagram"
                />,
                "href": "http://www.instagram.com/charlietcoder",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-instagram",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-behance"
                />,
                "href": "http://www.behance.net/chazhurricane",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-behance",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "data from ",
              <a
                href="https://www.cryptocompare.com/api/"
              >
                cryptocompare api
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "data from ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "cryptocompare api",
                "href": "https://www.cryptocompare.com/api/",
              },
              "ref": null,
              "rendered": "cryptocompare api",
              "type": "a",
            },
          ],
          "type": "p",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
